src/frameworks/CoreServices/src/LaunchServices/LSInfo.m

```diff
@@ -28,11 +28,15 @@
 __attribute__((visibility("hidden")))
 FMDatabaseQueue* getDatabaseQueue(void)
 {
-	static FMDatabaseQueue* db = nil;
-	if (!db)
-	{
+	static dispatch_once_t pred;
+    static FMDatabaseQueue* db;
+    dispatch_once(&pred, ^{
+	    db = nil;
+        if (!db)
```

Comment:
What's the point in setting it to `nil` and then immediately checking for it? 🤔

Reply:
I wrapped the existing code with the dispatch_once block, this could probably be omitted now that you point it out